cmds/task/fetchers.go

```diff
@@ -180,3 +181,29 @@ func runLog(credentials *tcclient.Credentials, args []string, out io.Writer, fla
 
 	return nil
 }
+
+// runAwait gets the status of run(s) of a given task.
+func runAwait(credentials *tcclient.Credentials, args []string, out io.Writer, flagSet *pflag.FlagSet) error {
```

Comment:
So this keeps polling the queue until the task is resolved..
I can see how it's useful, for small scripts or hacks where you want to wait for some thing to happen...

However, we don't have to be polling we have two other options:
 * Acquire pulse credentials from https://pulseguardian.mozilla.org/ and listen for events...
 * Implement a golang client for taskcluster-events (see: https://github.com/taskcluster/taskcluster-events)

For a quick update on what pulse is, take a look at:
https://wiki.mozilla.org/Auto-tools/Projects/Pulse
In short:
 * RabbitMQ is a pub-sub system, where you can publish messages and subscribe to messages,
 * Pulse is a RabbitMQ deployment with special rules, like:
   * We use topic exchanges meaning each message has a routing key on the form: "<key>.<words>.<...>" (dot separated)
   * All messages are considered public
   * Event publishers can only write to their own exchanges (anyone can read messages)
   * Access requires credential but these are easy to obtain
 * taskcluster-events is a simple gateway that allows for listening for pulse messages over a websocket without pulse credentials (so as to lower the entry bar, and be usable on the web)

It's a simple protocol, the idea is as follow:
1. Connect to `wss://events.taskcluster.net/v1//listen/websocket` using a websocket
2. Wait for message `{method: 'ready'}` from websocket
3. Send message `{method: 'bind', id: '<slugid.Nice()>', options: {exchange: '...', routingKeyPattern: '...'}}`
4. Wait for message `{method: 'bound', id: '<id-as-was-sent>'}`
5. Reload state of task from queue
6. Wait for message to arrive from websocket

If disconnected start from 1 again, as soon as you see the state of the task change to what you want, you're done...
You'll probably have to bind to each of the exchanges for task-completed, task-failed, etc... Notice that the routingKeyPattern could be: `primary.<taskId>.#` and then you'll only get message for said taskId (this is nice).

Information on pulse: https://wiki.mozilla.org/Auto-tools/Projects/Pulse
See queue exchanges: https://docs.taskcluster.net/reference/platform/taskcluster-queue/references/events

Implementations of client side for `taskcluster-events`: (yes, we have two)
* https://github.com/taskcluster/taskcluster-client-web/blob/master/src/WebListener.js
* https://github.com/taskcluster/taskcluster-client/blob/master/lib/weblistener.js

Server side implementation of `taskcluster-events`:
https://github.com/taskcluster/taskcluster-events/blob/master/events/socket.js

Note, if you wanted to do this, it would probably be smart to start a golang package (or subpackage) that focuses exclusively on implementing the `WebListener` (notice that this doesn't require any credentials to use).

Reply:
Sounds good. I threw this fairly quickly just to get it out. I'll look at it in more details after I land shell v1 :)